### ISSUE TITLE
`IString::intern("") != IString::default()`

### ISSUE THREAD

**Issue by scooter-dangle** [Diggsey/ijson#3]
The following fails on the second assertion with 0.1.1. I'd expect the assertions to always fail or succeed in tandem.

```rust
use ijson::IString;

#[test]
fn hmm() {
    let s1 = IString::intern("");
    let s2 = IString::default();
    assert_eq!(s1.as_str(), s2.as_str());
    assert_eq!(s1, s2);
}
```

**Comment by Diggsey** [comment]
Thanks for finding this! I've published 0.1.2 which contains a fix for this issue.

**Comment by scooter-dangle** [comment]
Awesome. Thanks!
